lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -48,6 +50,62 @@ def cmake_cache_filepath(name, value, comment=""):
     return 'set({0} "{1}" CACHE FILEPATH "{2}")\n'.format(name, value, comment)
 
 
+class Scheduler(enum.Enum):
+    LSF = enum.auto()
+    SLURM = enum.auto()
+    FLUX = enum.auto()
+
+
+def get_scheduler(spec: Spec) -> Optional[Scheduler]:
```

Comment:
It would be useful to push these to be class methods on `CachedCMakePackage` so that subclasses can easily call these without import weirdness

Reply:
Made the helper functions into class methods in aaadce69951